src/rocm_smi_device.cc

```diff
@@ -742,7 +742,8 @@ int Device::readDevInfoMultiLineStr(DevInfoTypes type,
     return 0;
   }
   // Remove any *trailing* empty (whitespace) lines
-  while (retVec->back().find_first_not_of(" \t\n\v\f\r") == std::string::npos) {
+  while (retVec->size() != 0 &&
```

Comment:
```suggestion
  while (!retVec->empty() &&
```
Also you can remove then the code from lines 741-743.

Reply:
Uh, this is not my contribution. Seems like @bill-shuzhou-liu force pushed a while ago? I just want to fix cmake stuff.